### PR TITLE
py-cf-units: new package

### PR DESCRIPTION
**Pull request by mjwoods**
Wrapper class to support Unidata/UCAR udunits2 and cftime calendar functionality.

Depends on #11890 .

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-cf-units/package.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyCfUnits(PythonPackage):
+    """Units of measure as required by the Climate and Forecast (CF)
+metadata conventions."""
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/py-cf-units/package.py]: Can you indent this line?

File: var/spack/repos/builtin/packages/py-cf-units/package.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    homepage = "https://scitools.org.uk"
+    url      = "https://pypi.io/packages/source/c/cf-units/cf-units-2.1.1.tar.gz"
+    git      = "https://github.com/SciTools/cf-units.git"
+
+    version('2.1.1', extension='tar.gz', sha256='fa0ef8efd84546e61088aa23e76ebbaf7043167dc3a7f35f34549c234b543530')
+    version('master', branch='master')
+
+    depends_on('python@3:', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+    depends_on('py-six',        type=('build', 'run'))
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/py-cf-units/package.py]: Extra spaces here

File: var/spack/repos/builtin/packages/py-cf-units/package.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+class PyCfUnits(PythonPackage):
+    """Units of measure as required by the Climate and Forecast (CF)
+metadata conventions."""
+
+    homepage = "https://scitools.org.uk"
+    url      = "https://pypi.io/packages/source/c/cf-units/cf-units-2.1.1.tar.gz"
+    git      = "https://github.com/SciTools/cf-units.git"
+
+    version('2.1.1', extension='tar.gz', sha256='fa0ef8efd84546e61088aa23e76ebbaf7043167dc3a7f35f34549c234b543530')
+    version('master', branch='master')
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/py-cf-units/package.py]: `master` version should come before numbered version.

File: var/spack/repos/builtin/packages/py-cf-units/package.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    """Units of measure as required by the Climate and Forecast (CF)
+    metadata conventions.
+    """
+
+    homepage = "https://scitools.org.uk"
+    url      = "https://pypi.io/packages/source/c/cf-units/cf-units-2.1.1.tar.gz"
+    git      = "https://github.com/SciTools/cf-units.git"
+
+    version('master', branch='master')
+    version('2.1.1', extension='tar.gz', sha256='fa0ef8efd84546e61088aa23e76ebbaf7043167dc3a7f35f34549c234b543530')
```
Comment (healther) [inline var/spack/repos/builtin/packages/py-cf-units/package.py]: just fyi: the extensions keyword should not be needed

Comment (citibeth) [reply var/spack/repos/builtin/packages/py-cf-units/package.py]: I'll merge once the `extension` keyword is removed.

**Comment by citibeth** [comment]
Ready to merge, once the button turns green.

**Pull request merged**